core/src/main/java/org/infinispan/persistence/sifs/FileProvider.java

```diff
@@ -210,9 +212,10 @@ protected FileChannel openChannel(int fileId) throws FileNotFoundException {
    protected FileChannel openChannel(File file, boolean create, boolean readSharedMeadata) throws FileNotFoundException {
       log.debugf("openChannel(%s)", file.getAbsolutePath());
 
-      FileChannel fileChannel = ATTEMPT_PMEM ? PmemUtilWrapper.pmemChannelFor(file, maxFileSize, create, readSharedMeadata) : null;
+      FileChannel fileChannel = ATTEMPT_PMEM && canTryPmem ? PmemUtilWrapper.pmemChannelFor(file, maxFileSize, create, readSharedMeadata) : null;
 
       if (fileChannel == null) {
+         canTryPmem = false;
```

Comment:
what if a user has multiple drives and some of them support pmem? As soon as it fails, you can no longer take advantage of pmem. 
Personally, I would just configure log4j and change `io.mashona` to `INFO` or `WARN`. 
Just my 2cents.

Reply:
just ignore me

Reply:
In case it isn't clear to others, this is what this change does. Each FileProvider is specific to a directory with SIFS. So after the first attempt (if mashona is on the classpath) for a directory if it fails, it will not attempt to use PMEM again.